### PR TITLE
adds config.json loader support and context query via handlebars templates

### PR DESCRIPTION
**Pull request by digi604**

### CODE REVIEW

File: ti/store/params.js [hunk 1]
```diff
@@ -31,7 +29,9 @@ export default {
     },
     mutations: {
         setParams (state, payload) {
-            state.params = payload
+            for (let a in payload) {
+                state[a] = payload[a]
```
Comment (torfeld6) [inline ti/store/params.js]: You might want to add a payload.hasOwnProperty(a) here.
Also is a Vue.set(state, a, payload[a]) might be needed.

File: ti/api/query.js [hunk 2]
```diff
@@ -1,86 +1,87 @@
...
+            }
+            if (cacheEnabled) {
+                cacheQuery(query, result)
+            }
+            resolve(result)
+        }).catch(reject)
     })
 }
 
+function contextQuery (query, context) {
```
Comment (torfeld6) [inline ti/api/query.js]: Can we make the contextQuery a new file? I think it's unecessary to import HandleBars, if I run a simple query

File: ti/api/query.js [hunk 3]
```diff
@@ -1,86 +1,87 @@
...
-                    reject(new Error(e.error))
-                }
-
-                for (let i = 0; i < e.results.length; i++) {
-                    let res = e.results[i];
-                    let obj = {};
+            db: 'pitcher',
+            pType: 'query',
+            query: query,
+        }).then(function (e) {
```
Comment (torfeld6) [inline ti/api/query.js]: Any particular reason for removing my formatting?

File: ti/api/query.js [hunk 4]
```diff
@@ -1,86 +1,91 @@
...
     return cacheEnabled
         && !query.toLowerCase().includes('delete')
         && !query.toLowerCase().includes('insert')
         && cache.hasOwnProperty(query)
-        && cache[query].time + cacheTimeout > Date.now();
+        && cache[query].time + cacheTimeout > Date.now()
 }
 
-function query(query, db = 'pitcher') {
+function query (query, db = 'pitcher') {
```
Comment (torfeld6) [inline ti/api/query.js]: Standard is no space after the function name

File: ti/api/query.js [hunk 5]
```diff
@@ -1,86 +1,91 @@
...
+let cacheEnabled = false
+let cacheTimeout = 500
 
-function cacheQuery(query, result) {
+function cacheQuery (query, result) {
     cache[query] = {
         result: result,
         time: Date.now()
-    };
+    }
```
Comment (torfeld6) [inline ti/api/query.js]: Let's stay with ; for now

File: ti/api/query.js (hunk 4 above)
Comment (digi604) [reply ti/api/query.js]: somehow newest intellij autoformater thinks this is a greate idea... disabled again

File: ti/store/params.js (hunk 1 above)
Comment (digi604) [reply ti/store/params.js]: done

File: ti/api/query.js (hunk 2 above)
Comment (digi604) [reply ti/api/query.js]: done

File: ti/api/query.js (hunk 5 above)
Comment (digi604) [reply ti/api/query.js]: semicolons are baaaack

**Pull request merged**